Catch all Swift exceptions in Mail::Send()

axometeam opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | Catch all Swift exceptions in Mail::Send() instead of throwing a Fatal Error
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-8913
| How to test?  | The forge ticket is full of details. But you can subscribe to mailalert with a wrong email (e.g.: foo.@bar.com) and create orders with the same product, which will trigger the hook "actionUpdateQuantity", then mailalerts module, then a 'Fatal error : Uncaught exception 'Swift_RfcComplianceException''

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/8763)
<!-- Reviewable:end -->

axometeam committed:
Catch all Swift exceptions in Mail::Send()

axometeam commented:
1.7 seems to have the same issue: http://forge.prestashop.com/browse/BOOM-4806

kpodemski commented:
btw. to someone who will review this, maybe we could create some Wrapper for this function? It's really annoying to use such a long function calls, a while ago i've created some super simple one, code here:

https://gist.github.com/kpodemski/498e1714e3188049932d8a013391d822

nsorosac commented:
UP please :-(

This can be merged in 1.6.1 and 1.7, do you need more info?

axometeam committed:
Allow duplicate logs for exceptions in Mail::Send

It's easier to debug Mail deliverability issues when the logs show real-time info.

The use case is "Sending an email > not receiving > going to the logs": if you already encountered the exception once, you won't get a log again, so you won't know if there is a problem or not.

axometeam commented:
Please :cry:

PierreRambaud commented on classes/Mail.php:
```diff
@@ -177,47 +177,48 @@ public static function Send($id_lang, $template, $subject, $template_vars, $to,
-                    return false;
-                }
+        try {
 
```
Useless line break :)

firespeed commented:
UP please !

eternoendless commented:
Thank you @axometeam

silvanoenrique commented:
> 1.7 seems to have the same issue: http://forge.prestashop.com/browse/BOOM-4806

Yes, but the solution is the same.
